stripe/_list_object.py

```diff
@@ -123,7 +141,13 @@ def __len__(self) -> int:
     def __reversed__(self) -> Iterator[T]:  # pyright: ignore (see above)
         return getattr(self, "data", []).__reversed__()
 
-    def auto_paging_iter(self) -> Iterator[T]:
+    def auto_paging_iter(self) -> SyncAsyncIterator[T]:
+        return SyncAsyncIterator(
+            self._auto_paging_iter(),
+            self._auto_paging_iter_async(),
+        )
```

Comment:
nit: might be worth adding a comment here or above on the `SyncAsyncIterator` class explaining how this supports multiple syntaxes